test/integ/test_key_pair_authentication.py

```diff
@@ -64,6 +64,49 @@ def test_get_token_from_private_key(input_account, expected_account):
     )
 
 
+@pytest.mark.skipolddriver
+def test_regionless_url_JWT_token_validity(request, db_parameters):
+
+    test_user = "admin"
+
+    db_config = {
+        "account": "amoghorgurl-keypairauth_test_alias.testdns",
+        "user": "admin",
+        "role": "ACCOUNTADMIN",
+        "timezone": "UTC",
+    }
+
+    db_config_with_pw = {
+        "account": "amoghorgurl-keypairauth_test_alias.testdns",
+        "user": "admin",
+        "password": "Password1",
+        "role": "ACCOUNTADMIN",
+        "timezone": "UTC",
+    }
+
+    cnx = snowflake.connector.connect(**db_config_with_pw)
+    cursor = cnx.cursor()
+    cursor.execute(
+        """
+    use role accountadmin
+    """
+    )
```

Comment:
You never close the connection or the cursor, which is why we prefer using the with clauses, they close things for you automatically
```suggestion
    with snowflake.connector.connect(**db_config_with_pw) as cnx:
        with cnx.cursor() as cursor:
            cursor.execute("use role accountadmin")
```